src/test/java/seedu/address/model/patient/BirthdayTest.java

```diff
@@ -0,0 +1,58 @@
+package seedu.address.model.patient;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.testutil.Assert.assertThrows;
+
+import org.junit.jupiter.api.Test;
+
+public class BirthdayTest {
+
+    @Test
+    public void constructor_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new Birthday(null));
+    }
+
+    @Test
+    public void constructor_invalidBirthdate_throwsIllegalArgumentException() {
+        String invalidBirthdate = "";
+        assertThrows(IllegalArgumentException.class, () -> new Birthday(invalidBirthdate));
+    }
+
+    @Test
+    public void isValidBirthdate() {
+        assertThrows(NullPointerException.class, () -> Birthday.isValidBirthdate(null));
+
+        // invalid birthdates
+        assertFalse(Birthday.isValidBirthdate("")); // empty string
+        assertFalse(Birthday.isValidBirthdate(" ")); // spaces only
+        assertFalse(Birthday.isValidBirthdate("22")); // less than 8 numbers
+        assertFalse(Birthday.isValidBirthdate("january")); // non-numeric
+        assertFalse(Birthday.isValidBirthdate("23071991")); // not in the format of dd/mm/yyyy
+        assertFalse(Birthday.isValidBirthdate("22 /01 / 1997")); // spaces within digits
+
+        // valid birthdates
+        assertTrue(Birthday.isValidBirthdate("21/01/1994")); // dd/mm/yyyy format
+        assertTrue(Birthday.isValidBirthdate("20/01/1974")); // dd/mm/yyyy format
+    }
```

Comment:
I like how you used a range of test cases!

Reply:
thanks